doc/Programming-guide/Using-Kaa-endpoint-SDKs/Java/index.md

```diff
@@ -150,12 +150,66 @@ The `newClient()` static method receives two mandatory arguments -- the platform
 
 The first argument can be either `DesktopKaaPlatformContext` for Java desktop or `AndroidKaaPlatformContext` for Android applications.
 
-In this example, the second argument is the default implementation of `KaaClientStateListener` -- `SimpleKaaClientStateListener`, that is solely used to log the client state changes.
+You can configure properties and `ExecutorContext` of `DesktopKaaPlatformContext` with help of it's constructors. `DesktopKaaPlatformContext` uses `FlexibleExecutorContext` implementation of `ExecutorContext` by default. You can change it by calling corresponding constructor with specific `ExecutorContext` implementation. 
+
+`FlexibleExecutorContext` has four threads group, which responsible for life cycle, callbacks, api and scheduling. Each of them has minimum and maximum threads amount and maximum idle time. Default minimum threads amount for all these groups is zero, maximum threads amount isn't limited and maximum idle time is 100 milliseconds. You can call corresponding constructor for change maximum threads amount or idle time (minimum threads amount is always zero). Except scheduled threads group, you can change only it's minimum threads amount. Recommended to use builder for this purpose. Here some examples for using non-empty constructor and using builder.
```

Comment:
- ... four **thread** group**s**, which responsible for life cycle, callbacks, **API** and scheduling. Each of these groups is implemented as separate thread pool (using ThreadPoolExecutor or its subclass).  
- ... and maximum **thread** idle time
- **By default** minimum treads amount ...
- You can call corresponding constructor **to limit** maximum threads amount
- **For** scheduled threads group, you can change
- **It is r**ecommended to use builder to construct the `FlexibleExecutorContext` instance with custom parameters.